unidist/core/backends/mpi/core/shared_object_store.py

```diff
@@ -0,0 +1,780 @@
+        self.service_shared_buffer = memoryview(service_buffer).cast("l")
+        # Set -1 to the service buffer because 0 is a valid value and may be recognized by mistake.
+        if mpi_state.is_monitor_process():
+            self.service_shared_buffer[:] = array(
+                "l", [-1] * len(self.service_shared_buffer)
+            )
+
+    def _parse_data_id(self, data_id):
+        """
+        Parse `DataID` object to pair of int.
+
+        Parameters
+        ----------
+        data_id : unidist.core.backends.common.data_id.DataID
+            The data identifier to be converted to a numerical form
+
+        Returns
+        -------
+        tuple
+            Pair of int that define the DataID
+        """
+        splited_id = str(data_id).replace(")", "").split("_")
+        return int(splited_id[1]), int(splited_id[3])
+
+    def _increment_ref_number(self, data_id, service_index):
+        """
+        Increment the number of references to indicate to the monitor that this data is being used.
+
+        Parameters
+        ----------
+        data_id : unidist.core.backends.common.data_id.DataID
+        service_index : int
+            The service buffer index.
+
+        Notes
+        -----
+        This function create `weakref.finalizer' with decrement function which will be called after data_id collecting.
+        """
+        if MPI.Is_finalized():
+            return
+        if service_index is None:
+            raise KeyError(
+                "it is not possible to increment the reference number for this data_id because it is not part of the shared data"
+            )
+        with SharedSignaler(self.service_win):
+            prev_ref_number = self.service_shared_buffer[
+                service_index + self.REFERENCES_NUMBER
+            ]
+            self.service_shared_buffer[service_index + self.REFERENCES_NUMBER] = (
+                prev_ref_number + 1
+            )
+            self.logger.debug(
+                f"Rank {communication.MPIState.get_instance().global_rank}: Increment references number for {data_id} from {prev_ref_number} to {prev_ref_number + 1}"
+            )
+        self.finalizers.append(
+            weakref.finalize(
+                data_id, self._decrement_ref_number, str(data_id), service_index
+            )
+        )
+
+    def _decrement_ref_number(self, data_id, service_index):
+        """
+        Decrement the number of references to indicate to the monitor that this data is no longer used.
+        When references count is 0, it can be cleared.
+
+        Parameters
+        ----------
+        data_id : unidist.core.backends.common.data_id.DataID
+        service_index : int
+            The service buffer index.
+
+        Notes
+        -----
+        This function is called in `weakref.finalizer' after data_id collecting.
+        """
+        # we must to set service_index in args because the shared_info will be deleted before than this function is called
```

Comment:
```suggestion
        # we must set service_index in args because the shared_info will be deleted before than this function is called
```
Which statetement does this phrase refer to?